internal/statestore/redis.go

```diff
@@ -20,29 +20,48 @@ import (
 	"io/ioutil"
 	"time"
 
-	"github.com/cenkalti/backoff"
-	"github.com/golang/protobuf/proto"
-	"github.com/gomodule/redigo/redis"
-	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"open-match.dev/open-match/internal/config"
-	"open-match.dev/open-match/pkg/pb"
-)
 
-const (
-	allTickets        = "allTickets"
-	proposedTicketIDs = "proposed_ticket_ids"
+	rs "github.com/go-redsync/redsync/v4"
+	rsredigo "github.com/go-redsync/redsync/v4/redis/redigo"
+	"github.com/gomodule/redigo/redis"
 )
 
 var (
 	redisLogger = logrus.WithFields(logrus.Fields{
 		"app":       "openmatch",
 		"component": "statestore.redis",
 	})
+
+	// this field is used to create new mutexes
+	redsync *rs.Redsync
 )
 
+// redisLocker is a wrapper over redsync mutex in order to let callers use it but not to expose redsync package outside
+type redisLocker struct {
+	mutex *rs.Mutex
+}
+
+// NewMutex returns a new distributed mutex with given name
+func (rb *redisBackend) NewMutex(key string) *redisLocker {
+	//TODO: make expiry duration configurable
+	m := redsync.NewMutex(key, rs.WithExpiry(5*time.Minute))
```

Comment:
@akremsa Can you add a prefix to the NewMutex:
```suggestion
	m := redsync.NewMutex("lock/"+key, rs.WithExpiry(5*time.Minute))
```

Reply:
Yes. 
The reason for this change is that miniredis store regular records and mutex keys in the same place and if we use the same key - backfill_id for both - miniredis can not acquire a lock during tests run.

Reply:
I assume that similar thing would happen for normal Redis also.

Reply:
did some tests - redis behaves the same